src/ucp/wireup/wireup.c

```diff
@@ -1039,33 +1078,47 @@ static void ucp_wireup_check_config_intersect(ucp_ep_h ep,
         ucs_assert(old_key->wireup_lane != UCP_NULL_LANE);
 
         /* set the correct WIREUP MSG lane in case of CM */
-        if ((new_key->wireup_lane != UCP_NULL_LANE) &&
-            (UCP_NULL_LANE !=
-             (reuse_lane = ucp_ep_config_matched_lane(old_key,
-                                                      new_key->wireup_lane,
-                                                      reuse_lane_map)))) {
-            new_uct_eps[new_key->wireup_lane] = ep->uct_eps[reuse_lane];
-            /* reset the UCT EP from the previous lane configuration to not
-             * destroy it, since it will be used by the new configuration */
-            ep->uct_eps[reuse_lane]           = NULL;
-        } else if (UCP_NULL_LANE !=
-                   (reuse_lane = reuse_lane_map[old_key->wireup_lane])) {
-            new_key->wireup_lane              = reuse_lane;
-        } else /* old wireup lane won't be re-used */ {
-            cm_wireup_ep                      = ucp_ep_get_cm_wireup_ep(ep);
-            new_key->wireup_lane              = new_key->cm_lane;
-            reuse_lane                        = old_key->wireup_lane;
-            /* save UCT EP from the previous WIREUP lane to the CM WIREUP
-             * auxiliary EP */
-            cm_wireup_ep->aux_ep              = ep->uct_eps[reuse_lane];
-            cm_wireup_ep->aux_rsc_index       = old_key->lanes[reuse_lane].rsc_index;
-            /* reset the UCT EP from the previous WIREUP lane to not
-             * destroy it, since it's not needed anymore in the new
-             * configuration, but will be used for WIREUP MSG */
-            ep->uct_eps[reuse_lane]           = NULL;
+        if (new_key->wireup_lane != UCP_NULL_LANE) {
+            reuse_lane = ucp_ep_config_find_lane_index(old_key,
+                                                       new_key->wireup_lane,
+                                                       reuse_lane_map);
+            if (reuse_lane != UCP_NULL_LANE) {
+                new_uct_eps[new_key->wireup_lane] = ep->uct_eps[reuse_lane];
+                /* reset the UCT EP from the previous lane configuration to not
+                 * destroy it, since it will be used by the new configuration */
+                ep->uct_eps[reuse_lane]           = NULL;
+            }
+        } else {
+            reuse_lane = reuse_lane_map[old_key->wireup_lane];
+            if (reuse_lane != UCP_NULL_LANE) {
+                new_key->wireup_lane              = reuse_lane;
+            } else /* old wireup lane won't be re-used */ {
+                cm_wireup_ep                      = ucp_ep_get_cm_wireup_ep(ep);
+                new_key->wireup_lane              = new_key->cm_lane;
+                reuse_lane                        = old_key->wireup_lane;
```

Comment:
is this done in order to avoid destroying rc qp on which we just got wireup_msg?
if yes, the logic should be different: the goad should not be to set new_key->wireup_lane (we can leave it as NULL), but to start discarding the old p2p lanes only after getting wireup_response (on client) or wireup_ack (on server). for this purpose, might need to push tmp_ep into cm_wireup_ep